Ex_10.py

```diff
@@ -0,0 +1,43 @@
+# This week’s exercise is going to be revisiting an old exercise (see Exercise 5), except require the solution in a different way.
+
+# Take two lists, say for example these two:
+
+# 	a = [1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89]
+# 	b = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
+# and write a program that returns a list that contains only the elements that are common between the lists (without duplicates). Make sure your program works on two lists of different sizes. 
+# Write this in one line of Python using at least one list comprehension. (Hint: Remember list comprehensions from Exercise 7).
+
+# The original formulation of this exercise said to write the solution using one line of Python, but a few readers pointed out that this was impossible to do without using sets that I had not yet discussed on the blog, 
+# so you can either choose to use the original directive and read about the set command in Python 3.3, or try to implement this on your own and use at least one list comprehension in the solution.
+
+# Extra:
+
+# Randomly generate two lists to test this
+
+import json
+import argparse
+parser = argparse.ArgumentParser()
+parser.add_argument("-f", type=str, nargs='?')
+args = parser.parse_args()
+
+data = None
+if args.f:
+    with open(args.f, 'r') as f:
+        data = json.load(f)
+if data == None:
+    a = [1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89]
+    b = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
+    c=[x for x in set(a) if x in (a and b) ]
```

Comment:
To działa ale nie tak jak sądzisz :)
```
In [1]: a = [1,2]

In [2]: b = [3,4]

In [3]: a and b
Out[3]: [3, 4]
```

Czyli `a and b` zwraca po prostu `b`. Tak na prawdę to co napisałaś wygląda tak:

```
c=[x for x in set(a) if x in b ]
```

Sumowanie list jest możliwe:
```
In [5]: a + b
Out[5]: [1, 2, 3, 4]
```

Rozwiązanie z list comprehension gdzie wstawiamy dodatkowy filtr w postaci `if x in b` jest bardzo wolne:

```
In [17]: a = [random.randrange(1, 500, 1) for i in range(1000)]

In [18]: b = [random.randrange(1, 500, 1) for i in range(2000)]

In [19]: timeit.timeit("c=list(set(a).intersection(b))", globals = globals())
Out[19]: 48.15213498599769

In [20]: timeit.timeit("c=[x for x in set(a) if x in b ]", globals = globals())
Out[20]: 956.6210397600007
```

Szybszym rozwiązaniem jest oparcie się całkowicie na typie set:
```
c = list( set(a).intersection(b) )
```